src/libraries/Microsoft.PowerFx.Connectors/ConnectorFunction.cs

```diff
@@ -975,8 +976,14 @@ internal static ConnectorType GetConnectorType(string valuePath, StringValue sv,
             return GetConnectorTypeInternal(compatibility, je);
         }
 
-        internal static ConnectorType GetConnectorType(string valuePath, StringValue sv, ConnectorCompatibility compatibility, out string name, out string displayName)
-        {
+        // Only called by CdpTabularService.GetSchema
+        internal static ConnectorType GetConnectorTypeAndTableCapabilities(string valuePath, StringValue sv, ConnectorCompatibility compatibility, out string name, out string displayName, out ServiceCapabilities tableCapabilities)
+        {                       
+            // There are some errors when parsing this Json payload but that's not a problem here as we only need x-ms-capabilities parsing to work
+            OpenApiReaderSettings oars = new OpenApiReaderSettings() { RuleSet = DefaultValidationRuleSet };
+            OpenApiSchema schema = new OpenApiStringReader(oars).ReadFragment<OpenApiSchema>(sv.Value, OpenApi.OpenApiSpecVersion.OpenApi2_0, out OpenApiDiagnostic _);            
+            tableCapabilities = schema.GetTableCapabilities();
```

Comment:
>schema.GetTableCapabilities(); [](http://example.com/codeflow?start=32&length=30)

Does PA get the capabilities from a swagger file, or from invoking a CDP endpoint?

Reply:
PA gets capabilities from Table service endpoint ($metadata.json).